File: STRIVER_SDE_SHEET/FindMissingNumber.java
package STRIVER_SDE_SHEET;

import java.util.Scanner;

public class FindMissingNumber {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int arr[] = new int[n];
        for(int i=0;i<n;i++){
            arr[i] = in.nextInt();
        }
        in.close();
        int hash[] = new int[n+1];
        for(int i=0;i<n;i++){
            hash[arr[i]]++;
        }
        int i = -1;
        for(i=1;i<=n;i++){
            if(hash[i]<=0){
                break;
            }
        }
        System.out.println(i);
    }
}

File: STRIVER_SDE_SHEET/Arrays_and_Hashing/FindMissingNumber.py
arr = []
n = int(input())
for i in range(n):
    arr.append(int(input()))

hash = [0]*(n+1)

for i in arr:
    hash[i]+=1
    
ans = None
for i in range(1,len(hash)+1):
    if hash[i]==0:
        ans=i
        break
print(ans) 